rust/azure_iot_operations_connector/src/file_mount_azure_device_registry.rs

```diff
@@ -1,4 +1,1110 @@
+                                            };
+
+                                        // Update the file mount map with the new assets
+                                        file_mount_map.update_assets(&device, &assets);
+                                    });
+                                }
+                                EventKind::Remove(event::RemoveKind::File) => { // Event signals the deletion of a device endpoint
+                                    debounced_event.paths.iter().for_each(|path| {
+                                        let device = match DeviceEndpointRef::try_from(path) {
+                                            Ok(device) => device,
+                                            Err(err) => {
+                                                log::warn!("Failed to parse device endpoint from path: {err:?}");
+                                                return;
+                                            }
+                                        };
+
+                                        // Remove the device endpoint from the file mount map
+                                        // When the device endpoint is removed, all the assets are 
+                                        // removed as well triggering the deletion token for each asset
+                                        file_mount_map.remove_device_endpoint(&device);
+                                    });
+                                }
+                                _ => { /* Ignore other events */ }
+                            }
+                        }
+                    }
+                    Err(err) => {
+                        // TODO: There should be a way for us to surface this error to the user
+                        for e in &err {
+                            log::error!("Error processing events from watcher: {e:?}");
+                        }
+                    }
+                }
+            },
+        ).map_err(ErrorKind::from)?;
+
+        // Start watching the file mount path for create, modify and remove events
+        debouncer
+            .watch(&mount_path, notify::RecursiveMode::NonRecursive)
+            .map_err(ErrorKind::from)?;
+
+        Ok(Self {
+            debouncer,
+            create_device_rx,
+        })
+    }
+
+    /// Receives a notification for a newly created device endpoint.
+    ///
+    /// Returns Some(([`DeviceEndpointRef`], [`AssetCreateObservation`])) if a notification is received or `None`
+    /// if there will be no more notifications (i.e. the channel is closed). This should not happen unless the
+    /// [`DeviceEndpointCreateObservation`] is dropped.
+    ///
+    /// The [`AssetCreateObservation`] should be used to receive notifications for asset creation events
+    /// associated with the device endpoint.
+    pub async fn recv_notification(
+        &mut self,
+    ) -> Option<(DeviceEndpointRef, AssetCreateObservation)> {
```

Comment:
should a device deletion token be returned here too, or is that information supposed to be inferred from the assetCreateObservation sending None? (if so, some more explicit doc around this would be good or maybe there's a better name for this Observation)

Reply:
It is in the case that the asset create observation sends none, I have those details in the documentation for that structure but will add here too